### PR TITLE
Add support for the Templates API endpoints

### PR DESCRIPTION
**Pull request by linzjax**
Basic support for the Klaviyo Templates API endpoints.

### CODE REVIEW

File: lib/klaviyo/apis/templates.rb [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    # Retrieve all email templates available.
+    # @param person_id [String] the id of the profile
+    # @return [JSON] A dictionary with a data property that contains an array of all the
+    #   email templates. Each entry is a separate email template object. If no email templates exist,
+    #   the resulting array will be empty. This request should never return an error.
+    def self.get_templates
+      v2_request(HTTP_GET, TEMPLATES)
+    end
+
+    # Creates a new email templates
```
Comment (njparadis) [inline lib/klaviyo/apis/templates.rb:17]: Small nit: template should be singular here, String is misspelled on the following line

File: lib/klaviyo/apis/templates.rb [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    RENDER = 'render'
+    SEND = 'send'
+
+    # Retrieve all email templates available.
+    # @param person_id [String] the id of the profile
+    # @return [JSON] A dictionary with a data property that contains an array of all the
+    #   email templates. Each entry is a separate email template object. If no email templates exist,
+    #   the resulting array will be empty. This request should never return an error.
+    def self.get_templates
+      v2_request(HTTP_GET, TEMPLATES)
```
Comment (njparadis) [inline lib/klaviyo/apis/templates.rb:14]: The Templates API is a V1 API, so I believe you should use the `v1_request` - have you tested these methods locally?

Comment (linzjax) [reply lib/klaviyo/apis/templates.rb:14]: Oh how embarrassing. I spent a good deal of time Thursday trying to figure out why my integration wasn't working in my app, turns out I just hadn't tested thoroughly. Yikes. I'll fix that...

Comment (linzjax) [reply lib/klaviyo/apis/templates.rb]: I've pushed the change, but I'll let you know once I've fully tested it.

**Comment by linzjax** [comment]
I'm going to close this for now since I'm running into some issues. I'll reopen sometime in the future

**Pull request closed without merging**